### PR TITLE
Fix 8gpu PP failure due to 2D DCP disablement

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #460

DCP recently added safeties to avoid using it for 2D/3D since strided
sharding (a feature needed for safe 2D/3D resharding) is not ready yet.

PP uses DCP to load a seed checkpoint.  Disabling the safety mechanism
is enough to make 3D/PP still work (for the case where we train from the
beginning or do not re-shard.

(Resharding refers to saving a checkpoint from one world
size/parallelism config and loading/resuming under a different one).

### CODE REVIEW

**Review by fegin — APPROVED** [review]
LGTM, we should be able to remove this after @XilunWu WIP PRs landed.

**Comment by tianyu-l** [comment]
@wconstab @fegin quick sync: do we want to land this fix, or just wait for Xilun's change?

**Comment by wconstab** [comment]
yea i forgot about this, and wanted to get a review from @fegin first.  I think we should land it for now and remove it when Xilun's fix lands.

How do we make sure we don't forget to remove this...

**Comment by tianyu-l** [comment]
> How do we make sure we don't forget to remove this...

tracked in #486

**Pull request merged**
